VersionMiner/GitHubAction.cs

```diff
@@ -0,0 +1,198 @@
+            var versionKeys = inputs.VersionKeys.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+            if (versionKeys.Length <= 0)
+            {
+                throw new NoVersionFoundException("No version keys supplied for the 'version-keys' input.");
+            }
+
+            _gitHubConsoleService.Write("version keys validated.", true);
+            _gitHubConsoleService.BlankLine();
+
+            _gitHubConsoleService.Write("✔️️Pulling version from file . . . ");
+            foreach (var versionKey in versionKeys)
+            {
+                var keyExists = _xmlParserService.KeyExists(xmlData, versionKey, inputs.CaseSensitiveKeys);
+
+                var keyValue = keyExists
+                    ? _xmlParserService.GetKeyValue(xmlData, versionKey)
+                    : string.Empty;
+                keyValues.Add(keyValue);
+            }
+
+            _gitHubConsoleService.Write("version pulled from file.", true);
+            _gitHubConsoleService.BlankLine();
+
+            /* If the action should fail on key value mismatch,
+             * collect all of the values for comparison.
+            */
+            if (inputs.FailOnKeyValueMismatch)
+            {
+                if (keyValues.TrueForAll(v => v == keyValues[0]) is false)
+                {
+                    var exceptionMsg = "All values must match.";
+                    exceptionMsg += "This failure only occurs if the 'fail-on-key-value-mismatch' action input is set to 'true'.";
+
+                    throw new ValuesMismatchException(exceptionMsg);
+                }
+            }
+
+            // Just get the first value that is not empty
+            var version = keyValues.Find(v => !string.IsNullOrEmpty(v)) ?? string.Empty;
+
+            if (inputs.FailWhenVersionNotFound && string.IsNullOrEmpty(version))
+            {
+                var exceptionMsg = "No version value was found.";
+                exceptionMsg += $"{Environment.NewLine}If you do not want the GitHub action to fail when no version is found,";
+                exceptionMsg += "set the 'fail-when-version-not-found' input to a value of 'false'.";
+
+                throw new NoVersionFoundException(exceptionMsg);
+            }
+
+            _gitHubConsoleService.StartGroup("Version Miner Outputs");
+            _gitHubConsoleService.WriteLine($"version: {version}");
+            _gitHubConsoleService.EndGroup();
+            _gitHubConsoleService.BlankLine();
+
+            _actionOutputService.SetOutputValue(VersionOutputName, version);
+        }
+        catch (Exception e)
+        {
+            onError(e);
+        }
+
+        onCompleted();
+    }
+
+    /// <inheritdoc/>
+    public void Dispose() => Dispose(true);
+
+    /// <summary>
+    /// <inheritdoc cref="IDisposable.Dispose"/>
+    /// </summary>
+    /// <param name="disposing">True to dispose of managed resources.</param>
```

Comment:
Reword
```suggestion
    /// <param name="disposing">Disposes managed resources.</param>
```

Reply:
This is explaining what the `disposing` bool parameter does.  Not what the method does.
Basically, the `boolean` value of the `disposing` parameter must be `true` for the managed resources to be disposed of.

Does this mean I would still need to refactor the text?